### PR TITLE
logtalk 3.00.2

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

File: Library/Formula/logtalk.rb [hunk 1]
```diff
@@ -1,33 +1,54 @@
...
+
+    lgt = if build.with?("swi-prolog")
+            "swilgt"
+          elsif build.with?("gnu-prolog")
+            "gplgt"
+          else
+            "yaplgt"
+          end
+
+    pipe_output("#{lgt} #{home}/examples/ack/ack.lgt 2>/dev/null", "")
```
Comment (mikemcquaid) [inline Library/Formula/logtalk.rb]: :heart_eyes:

**Comment by bfontaine** [comment]
The build fails because the bot fails to build [`yap`](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/yap.rb).

**Comment by mikemcquaid** [comment]
@bfontaine Mind taking a look at that? A very quick look looks like it's failing to build docs which we could disable if necessary. Thanks!

File: Library/Formula/logtalk.rb [hunk 2]
```diff
@@ -1,33 +1,54 @@
...
+  EOS
+  end
+
+  test do
+    user_setup = testpath/"test-logtalk"
+    home = "#{share}/logtalk-#{version}-stable"
+    ENV["LOGTALKHOME"] = home
+    ENV["LOGTALKUSER"] = user_setup
+    system "#{bin}/logtalk_user_setup"
+    assert Dir.exist? user_setup
```
Comment (jacknagel) [inline Library/Formula/logtalk.rb]: `Dir.exist?` doesn't exist in Ruby 1.8, so use `File.directory?` here.

File: Library/Formula/logtalk.rb (hunk 1 above)
Comment (jacknagel) [inline Library/Formula/logtalk.rb]: `pipe_output` [makes no assertions](https://github.com/Homebrew/homebrew/blob/700635915dcd0cbc412cd7aab8aac177f3d8acbc/Library/Homebrew/formula_assertions.rb#L35-L41), it only returns the output of running the command.

For example, `pipe_output("exit 1", "")` succeeds.

File: Library/Formula/logtalk.rb (hunk 2 above)
Comment (bfontaine) [inline Library/Formula/logtalk.rb]: @jacknagel Thanks, fixed.

File: Library/Formula/logtalk.rb (hunk 1 above)
Comment (bfontaine) [inline Library/Formula/logtalk.rb]: @jacknagel Thanks, I didn’t know. Is there any reason it doesn’t check `$?`?

**Comment by bfontaine** [comment]
@mikemcquaid I will look into that.

**Comment by bfontaine** [comment]
@mikemcquaid I can’t reproduce the build failure on my mac (OSX 10.10.1)

**Comment by mikemcquaid** [comment]
> @mikemcquaid I can’t reproduce the build failure on my mac (OSX 10.10.1)

Maybe depends on Java 8 or something?

**Comment by bfontaine** [comment]
Tried `depends_on :java => ["1.8", :build]` without success. I’ll update this post with my progress.

* The build is failing on a sub-package called “JPL”, an interface between Java and Prolog. It can’t explicitly be disabled at the `./configure` step (there’s no `--disable-jpl` option), but from what I’ve seen in `configure`, removing `--with-java` disables it.
* If I remove `--with-java` the compilation [fails](http://bot.brew.sh/job/Homebrew%20Pull%20Requests/20034/version=mountain_lion/console) on something that seems completely unrelated:

         (…)
         gcc-4.9 -fstrict-aliasing -freorder-blocks -fsched-interblock -Wall -Wstrict-aliasing=2 -c -DBP_FREE -O3 -fomit-frame-pointer -Wall -Wstrict-prototypes -Wmissing-prototypes -g -O2 -I/usr/local/opt/gmp/include -DCUT_C=1  -DMYDDAS_ODBC -DCOROUTINING=1 -DRATIONAL_TREES=1 -DDEPTH_LIMIT=1 -DTHREADS=1 -D_GNU_SOURCE -DTABLING=1 -DHAVE_CONFIG_H -D_YAP_NOT_INSTALLED_=1 -I/usr/local/opt/gmp/include -I/usr/local/opt/readline/include  -I. -I./H -I./OPTYap -I./BEAM -I./MYDDAS C/gprof.c -o gprof.o
          In file included from /usr/include/sys/ucontext.h:33:0,
                           from C/gprof.c:75:
          C/gprof.c: In function 'LeftRotate':
          C/gprof.c:202:20: error: expected identifier or '(' before 'void'
             rb_red_blk_node* nil=ProfilerNil;
                              ^
          C/gprof.c:202:20: error: expected ')' before numeric constant
             rb_red_blk_node* nil=ProfilerNil;

**Comment by bfontaine** [comment]
@mikemcquaid The build has multiple issues, but on Mavericks it [fails saying that Java 1.7 should be installed](http://bot.brew.sh/job/Homebrew%20Pull%20Requests/22186/version=mavericks/testReport/junit/brew-test-bot/mavericks/install_yap/) :disappointed:

**Comment by mikemcquaid** [comment]
@bfontaine Should be fixed now, thanks. Rerunning.

**Comment by bfontaine** [comment]
@mikemcquaid Still the same error :/

> yap: Java 1.7 is required to install this formula.
> You can install with Homebrew Cask:
>   brew install Caskroom/cask/java
>
> You can download from:
>  http://www.oracle.com/technetwork/java/javase/downloads/index.html
> Error: An unsatisfied requirement failed this build.

**Comment by bfontaine** [comment]
@mikemcquaid Thoughts on the issue above?

**Comment by mikemcquaid** [comment]
If this can't work with the current stable version of Java I think it might be a candidate for boneyarding :(

**Comment by bfontaine** [comment]
Thanks. logtalk 3.00.2 is out, I’m trying it and if nothing works I’ll boneyard.

**Comment by bfontaine** [comment]
I finally found a way to get a working `yap` but the formula is messy. At least it’s easier to clean a working formula than to repair a broken one.

File: Library/Formula/yap.rb [hunk 3]
```diff
@@ -1,38 +1,84 @@
...
+  depends_on "homebrew/versions/gcc49"
+  depends_on :java => "1.8+"
 
   fails_with :clang do
     cause "uses variable-length arrays in structs, which will never be supported by clang"
   end
 
+  stable do
+    # http://permalink.gmane.org/gmane.comp.ai.prolog.yap.general/1790
+    patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/yap.rb]: Can you add a wee comment explaining why this is needed?

**Comment by mikemcquaid** [comment]
audit failing for yap --devel: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/27287/version=yosemite/testReport/junit/brew-test-bot/yosemite/audit___devel_yap/

File: Library/Formula/yap.rb (hunk 3 above)
Comment (bfontaine) [inline Library/Formula/yap.rb]: Yes, I’m still experimenting here, this PR is not merge-ready (and it fails on devel).

**Comment by bfontaine** [comment]
Closing this as `logtalk` was updated to 3.00.4 by #41434. I’ll fix or boneyard `yap` in another PR.

**Pull request closed without merging**